### ISSUE TITLE
ArrayField | ArrayFieldRemoveButton |Delete array item not working properly 

### ISSUE THREAD

**Issue by divan1920** [saas-js/saas-ui#210]
### Description

Deleting an item using `ArrayFieldRemoveButton` deletes the item from the view, but it does not alter the `formState`. 

## Ex. 
Suppose you have passed the default values to pre-populate the form with existing data ( in the case of edit ). You delete an item and then add a new item.
### Expected behavior: 
- The existing item will be removed, and an empty item will be added.

### Current behavior:
- The existing item gets removed, when added new item is pre-polutated with the default values of deleted item.

## Code from SaaS UI Docs
```
import { SubmitButton } from '@saas-ui/react'
import { Form } from '@saas-ui/forms/yup'

import * as yup from 'yup'

const issueSchema = yup.object().shape({
  title: yup.string().required().label('Title'),
})

const schema = issueSchema.concat(
  yup.object().shape({
    issues: yup.array().min(1).max(4).of(issueSchema),
  })
)

export default function CreateIssue() {
  return (
    <Form
      schema={schema}
      defaultValues={{
        title: '',
        issues: [{ title: 'issue1' },{ title: 'issue2' },{ title: 'issue3' },],
      }}
      onSubmit={() => Promise.resolve()}
    >
      {({ Field, ArrayField }) => (
        <FormLayout>
          <Field name="title" label="Title" />

          <ArrayField name="issues" label="Sub-issues">
            <Field name="issues.$.title" placeholder="Sub-issue title" />
          </ArrayField>

          <SubmitButton>Create todo</SubmitButton>
        </FormLayout>
      )}
    </Form>
  )
}
```
##Video explaining the issue, running the above code

https://github.com/saas-js/saas-ui/assets/140388383/d4d3c6f0-a30b-4388-98b8-8ae0e0b26b15

### Saas UI Version

2.8.2

### Chakra UI Version

2.6.0

### Operating System

- [X] macOS
- [X] Windows
- [X] Linux

**Comment by divan1920** [comment]
Relates to https://discord.com/channels/937049452349382677/1213005880027512852/1213005880027512852

**Comment by Pagebakers** [comment]
Thanks for opening the issue. I suspect the solution will be quite straightforward but need to test it.

For now you can use `setValue` manually to clear any existing values as a workaround.

**Comment by Munaf-Divan** [comment]
> To make sure this works as you expect you need to set empty default values on the ArrayField, otherwise it will fallback to the defaultValues of the form.
> 
> ```ts
>             <ArrayField
>               name="issues"
>               label="Sub-issues"
>               defaultValue={{
>                 title: '',
>               }}
>             >
> ```

No, this does not solve the problem; you might have misunderstood the bug here.
We can add an infinite number of items even if the default values don't have the same number. The solution you have provided will work if I add one item, then it will take this value, but what if I add one more.
The way you are suggesting is like I need to many default values to let user add empty row.

**Comment by Munaf-Divan** [comment]
This is quite a different code snippet, but this does solve the problem of adding an empty item in the array. But it would have been best if you could support it by default.

```
       <Button
          aria-label='add-delivery-spot'
          size='sm'
          variant='link'
          isDisabled={!isValid} // disable button when user has not entered a valid field values
          leftIcon={<HiPlusSm boxSize='14px' />}
          onClick={() => {
            append(
              { name: '', producer_number: '', image: '' },
              { shouldFocus: false },
            ); // add a row into the array and don't focus any field
          }}
          {...props}
        >
          {t('labels.addDeliverySpot')}
        </Button>
 ```

**Comment by Pagebakers** [comment]
Hmm yeah maybe I understand you incorrectly, but calling the append with empty values should have the same effect as

```ts
   <ArrayField
              name="issues"
              label="Sub-issues"
              defaultValue={{
                name: '', producer_number: '', image: ''
              }}
            >
```

Now when you append a new row to the array, it will default to these values. without having to add them as the first parameter of `append`.
